src/services/contract-service.js

```diff
@@ -239,6 +277,7 @@ class ContractService {
         .send(opts)
         .on('receipt', resolve)
         .on('confirmation', confirmationCallback)
```

Comment:
Won't we have same issue here mentioned in the deploy method that confirmationCallback does not get triggered ? Are we relying on this callback anywhere in our DApp or origin-js codebase ? If yes should we fix it, if no can we at least add a comment to warn caller that this is currently not working ?

Reply:
Interestingly when doing an Ethereum transaction confirmation callback is working, but when deploying a contract it is not. That is why the workaround is implemented only in the `deploy` function a few lines above.

Reply:
Got it. Thanks for being thorough.